### PR TITLE
Vagrant 1.7 and latest deps

### PR DESCRIPTION
**Pull request by electrofelix**
Add's support for vagrant 1.7, updates the lock file to recent dependencies, includes dependencies for Windows, ensures that chef-zero is run with the same ruby as vagrant so that "#!/usr/bin/env ruby" doesn't trigger errors on Windows and makes it easier to bump patch release numbers while allowing building of pre-release gems with identifiable versions.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -27,47 +27,31 @@ PATH
 GEM
```
Comment (miguelaferreira) [inline Gemfile.lock]: Should this file be committed to the repo?

I always add *.lock to .gitignore to prevent accidentally committing.

**Comment by miguelaferreira** [comment]
Hi @electrofelix 

I've been following this repository for a while now and I've seen several good PRs (this one included) simply be left here without any feedback or prospect of being merged.

Meanwhile, I've forked it and merged the PRs that I found good and useful.
Although, I don't particularly like having done that, it seemed a way to make the situation workable.

Since you made this PR recently, I assume you are actively interested in using the plugin.
I am too, but I don't see how to proceed, since I can't seem to get in contact with the original maintainer.

**Comment by andrewgross** [comment]
Hey every, I just got @miguelaferreira's email.  Sorry for not responding to these issues.  I do not actively use this plugin anymore, so I am working to transfer the project so you can all continue developing it.

Sorry for delaying project so much.

File: Gemfile.lock (hunk 1 above)
Comment (electrofelix) [inline Gemfile.lock]: I think that it's a case of it depends, if you have something that is more an application than a library, then perhaps you should. Certainly I think this plugin with it's dependencies is in a grey area.

Comment (miguelaferreira) [inline Gemfile.lock]: I guess that for an application you would package the entire bundle, and in that case you still wouldn't need the lock file. You would build your application and package the "binaries".

But, anyway, this is a plugin for vagrant (which in turn is an application). Right?

**Comment by miguelaferreira** [comment]
Hi @electrofelix 

I've looked through your changes and it LGTM.
However, since some of the older PRs have been merged, this one does not merge cleanly anymore. 
Would you be so kind as to rebase your PR?

File: Gemfile.lock (hunk 1 above)
Comment (electrofelix) [inline Gemfile.lock]: I'm open to it being either way, so whichever way you decide to go with is fine with me.

I suspect the reason some plugins adopted this is that you can't pin the dependencies for all vagrant plugins with vagrant, so the standard behaviour of installing plugins without a Gemfile.lock will pull in the latest gems allowed at install time. Looking at other plugins I'm using it does seem to depend on the project.

With Gemfile.lock
vagrant-lxc
vagrant-cachier
vagrant-chef-zero

Without:
vagrant-hostmanager
vagrant-libvirt
vagrant-mutate
vagrant-omnibus
vagrant-proxyconf

Comment (miguelaferreira) [inline Gemfile.lock]: I see the point, and for sure we need control over dependency versions. But we can specify version numbers in the Gemfile right?

I mean the Gemfile is the source for getting Gemfile.lock, and as long as the source is stable, then the generated lock file should always be the same.

**Comment by miguelaferreira** [comment]
LGTM

**Pull request merged**